testing/snapshot.ts

```diff
@@ -0,0 +1,205 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+import { fromFileUrl, join, parse, toFileUrl } from "../path/mod.ts";
+import { ensureFile, ensureFileSync } from "../fs/mod.ts";
+import { bold, green, red } from "../fmt/colors.ts";
+import { AssertionError, equal } from "./asserts.ts";
+import { buildMessage, diff, diffstr } from "./_diff.ts";
+import { format } from "./_format.ts";
+
+const CAN_NOT_DISPLAY = "[Cannot display]";
+const SNAPSHOT_DIR = "__snapshots__";
+
+type AssertSnapshotContext = {
+  snapshotPath: string | null;
+  teardownRegistered: boolean;
+  currentSnapshot: Map<string, string> | null;
+  updatedSnapshot: Map<string, string>;
+  snapshotCounts: Map<string, number>;
+  snapshotUpdatedCount: number;
+};
+
+let _assertSnapshotContext: AssertSnapshotContext;
+
+/**
+ * Write updates to the snapshot file.
+ *
+ * @param context Assert snapshot context
+ */
+function writeSnapshotFileSync(context: AssertSnapshotContext) {
+  const buf = ["export const snapshot = {};"];
+  function escapeStringForJs(str: string) {
+    return str
+      .replace(/\\/g, "\\\\")
+      .replace(/`/g, "\\`")
+      .replace(/\$/g, "\\$");
+  }
+  context.updatedSnapshot.forEach((value, key) => {
+    let formattedValue = escapeStringForJs(value);
+    formattedValue = formattedValue.includes("\n")
+      ? `\n${formattedValue}\n`
+      : formattedValue;
+    const formattedKey = escapeStringForJs(key);
+    buf.push(`\nsnapshot[\`${formattedKey}\`] = \`${formattedValue}\`;`);
+  });
+  ensureFileSync(context.snapshotPath as string);
+  Deno.writeTextFileSync(context.snapshotPath as string, buf.join("\n"));
```

Comment:
Because `snapshotPath` is now file url, I think we need to convert it back to file path

Reply:
Done 🙂